packages/polygon/src/Polygon.ts

```diff
@@ -0,0 +1,58 @@
+import { type BridgeActionParams, compressJson } from '@rabbitholegg/questdk'
+import { POLYGON_BRIDGE_ABI_FUNCS } from './abi.js'
+import { POLYGON_CHAIN_ID, CHAIN_ID_ARRAY } from './chain-ids.js'
+import {
+  PolygonTokens,
+  ETH_ADRESS_MAINNET,
+} from './supported-token-addresses.js'
+import { MAINNET_BRIDGE } from './contract-addresses.js'
+// If you're implementing swap or mint, simply duplicate this function and change the name
+export const bridge = async (bridge: BridgeActionParams) => {
+  // This is the information we'll use to compose the Transaction object
+  const { sourceChainId, contractAddress, tokenAddress, amount, recipient } =
+    bridge
+
+  // L2 Transactions are the same for ETH and ERC20
+  if (sourceChainId === POLYGON_CHAIN_ID) {
+    return compressJson({
+      chainId: sourceChainId, // The chainId of the source chain
+      to: contractAddress || tokenAddress, // on Polgon the contract that handles withdrawals is the token contract
+      input: {
+        $abi: POLYGON_BRIDGE_ABI_FUNCS,
+        amount: amount,
+      },
+    })
+  }
+  // Handle L1 ETH tx
+  if (tokenAddress === ETH_ADRESS_MAINNET) {
+    return compressJson({
+      chainId: sourceChainId, // The chainId of the source chain
+      to: contractAddress || MAINNET_BRIDGE, // on Polgon the contract that handles withdrawals is the token contract
+      value: amount,
+      input: {
+        $abi: POLYGON_BRIDGE_ABI_FUNCS,
+        user: recipient,
+      },
+    })
+  }
+  // Handle L1 ERC20 tx
+  // The deposit function takes a bytes chunk as the last argument, and the amount is encoded in that value
+  return compressJson({
+    chainId: sourceChainId, // The chainId of the source chain
+    to: contractAddress || MAINNET_BRIDGE, // on Polgon the contract that handles withdrawals is the token contract
+    input: {
+      $abi: POLYGON_BRIDGE_ABI_FUNCS,
+      user: recipient,
+      rootToken: tokenAddress,
+    },
+  })
+}
+
+export const getSupportedTokenAddresses = async (_chainId: number) => {
```

Comment:
Does this need to be an `async` function?

Reply:
The type definition specifies `async`, but this function does not need to be.
```
export interface IActionPlugin {
  pluginId: string
  getSupportedChainIds: () => Promise<number[]>
  getSupportedTokenAddresses: (chainId: number) => Promise<Address[]>
  bridge: (
    params: BridgeActionParams,
  ) => Promise<TransactionFilter> | Promise<PluginActionNotImplementedError>
  swap: (
    params: SwapActionParams,
  ) => Promise<TransactionFilter> | Promise<PluginActionNotImplementedError>
  mint: (
    params: MintActionParams,
  ) => Promise<TransactionFilter> | Promise<PluginActionNotImplementedError>
}
```

What do you think, better to just drop it in the definition still? I'm ambivalent, as long as it's consistent how we handle it.

Reply:
Ok cool, lets keep it consistent then! `async` it is